### PR TITLE
-Wdeprecated-register: Silent the warning when building with a non C++17 compiler

### PR DESCRIPTION
**Pull request by sylvestre**
Now, fixing the warning!

### CODE REVIEW

File: lib/lz4.c [hunk 1]
```diff
@@ -88,10 +88,18 @@
 /*
  * register is ignored when the code built with a C++-17 compiler
  * Remove the keyword when built with C++-17 to silent the warning
+ * Ignore the warning with other C++ versions
  */
-#if defined(__cplusplus) &&  __cplusplus > 201402L
+#ifdef __cplusplus
+#if __cplusplus > 201402L
```
Comment (Cyan4973) [inline lib/lz4.c]: minor: try to indent inner `#if`

**Comment by sylvestre** [comment]
I saw the CI failure. 
Another solution would be to disable the register keyword for all C++ . I am not sure it is useful with C++:
http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2015/n4340.html
http://en.cppreference.com/w/cpp/keyword/register

**Comment by Cyan4973** [comment]
That's probably true.
To be complete, this hypothesis should be checked (benchmarked) against a variety of compilers to ensure it's not detrimental for some targets.

**Comment by Cyan4973** [comment]
OK, let's remove `register` symbol,
it's likely the simplest thing to do at this stage.

**Pull request closed without merging**